plugin/pkg/proxy/connect.go

```diff
@@ -117,6 +118,20 @@ func (p *Proxy) Connect(ctx context.Context, state request.Request, opts Options
 	for {
 		ret, err = pc.c.ReadMsg()
 		if err != nil {
+			// If the error is an overflow, we probably have a misbehaving upstream
+			// that is sending out-of-spec (>512 byte) UDP responses without an eDNS0
+			// OPT RR. Set the truncate bit and return the result - this will make the
+			// client retry over TCP (if that's supported) or at least receive a clean
+			// error. The connection is still good so we break before the close.
+			if proto == "udp" && strings.Contains(err.Error(), "overflow") {
+				newRet := state.Req.Copy()
+				newRet.AuthenticatedData = false
```

Comment:
Please add a note here for why AD bit is cleared.  e.g. something like ...

Clear AD bit in case request had set the AD bit.  The empty response is not authenticated.